packages/cozy-client/src/mock.js

```diff
@@ -27,32 +36,44 @@ const mockedQueryFromMockedRemoteData = remoteData => qdef => {
 }
 
 /**
- * Creates a client suitable for use in tests
+ * Creates a client suitable for use in tests or documentation
  *
- * - client.{query,save} are mocked
- * - client.stackClient.fetchJSON is mocked
+ * If isDocumentation is false:
+ *   - client.{query,save,saveAll} are mocked
+ *   - client.stackClient.fetchJSON is mocked
  *
- * @param  {object} options Options
- * @param  {object} [options.queries] Prefill queries inside the store
- * @param  {object} [options.remote] Mock data from the server
- * @param  {object} [options.clientOptions] Options passed to the client
+ * If isDocumentation is true:
+ *   - All jest mock is bypassed
+ *
+ * @param {object} options Options
+ * @param {object} [options.queries] Prefill queries inside the store
+ * @param {object} [options.remote] Mock data from the server
+ * @param {object} [options.clientOptions] Options passed to the client
+ * @param {boolean} [options.isDocumentation] Define if used for documentation or test
  * @returns {CozyClient}
  */
-const createMockClient = ({ queries, remote, clientOptions }) => {
+const createMockClient = ({
+  queries,
+  remote,
+  clientOptions,
+  isDocumentation = false
+}) => {
```

Comment:
I dislike this kind of API. We're creating a client, with all arguments related to the client. I don't think that adding a `isDocumentation` there is not a good idea. 

What do you think of instead of adding a new boolean attribute, we add a new object containing the method we want to mock? 

Something like: createMockClient({queries, remote, clientOptions, mockedMethod: ... ?

Also, why in UI you need to fillQueryInsideClient but you don't want to mock client.query() to read from there?

Reply:
ping @Merkur39